timescaledb/tutorials/grafana/visualizations/time-series.md

```diff
@@ -0,0 +1,194 @@
+
+1.  In Grafana, create a new panel and add this query:
+
+    ```sql
+    SELECT time_bucket('$bucket_interval', time) AS time,
+    	AVG(price) AS price
+    FROM stocks_real_time
+    WHERE symbol = '$symbol'
+        AND time >= $__timeFrom()::timestamptz AND time < $__timeTo()::timestamptz
+    GROUP BY time_bucket('$bucket_interval', time);
+    ```   
+
+1.  With a `bucket_interval` of 30 minutes and a date range of `Last 30 days`, Grafana
+    returns this graph:
+
+    <img class="main-content__illustration" src="https://assets.timescale.com/docs/images/tutorials/visualizations/time-series/time-bucket-graph.png" alt="Screenshot of the time-series graph produced by Grafana using the `time_bucket` hyperfunction. The graph represents the price of AMD stock in the past 30 days with an interval of 30 minutes."/>
+
+    Because the stock market is only open from 9:30&nbsp;AM to 4:00&nbsp;PM on weekdays, 
+    there are large gaps in our dataset where no data is present. Grafana 
+    automatically connects the last non null to the nearest other non null value.
+    In this case, the `time_bucket` hyperfunction doesn't return a row if there 
+    is no data in the bucket. This stops you from using the `Connect null values`
+    option in the options panel.
+
+1.  In your query, replace `time_bucket` with `time_bucket_gapfill`.
+   
+    ```SQL
+    SELECT time_bucket_gapfill('$bucket_interval', time) as time,
+	    AVG(price) as price 
+    FROM stocks_real_time
+    WHERE symbol = '$symbol'
+        AND time >= $__timeFrom()::timestamptz AND time < $__timeTo()::timestamptz
+    GROUP BY time_bucket_gapfill('$bucket_interval', time);
+    ```
+
+    You can use [`time_bucket_gapfill`][time-bucket-gapfill] to circumvent the 
+    issue of `time_bucket` not returning rows when the bucket is empty. Usually,
+    `time_bucket_gapfill` is used in combination with [`LOCF`][locf] 
+    ("Last Observation Carried Forward") but if you omit this function, 
+    `time_bucket_gapfill` returns a null value which you can use to connect 
+    values based on a threshold in Grafana.
+
+1.  Set `Connect null values` in the options panel to `Threshold` with a value
+    of `24h`.
+
+    <img class="main-content__illustration" src="https://assets.timescale.com/docs/images/tutorials/visualizations/time-series/connect-null-values.png" alt="Screenshot of the 'Connect null values' in the time-series options panel. The selected value is 'Threshold' and it has a value of less than 24 hours."/>
+
+    . Grafana returns this graph:
+
+    <img class="main-content__illustration" src="https://assets.timescale.com/docs/images/tutorials/visualizations/time-series/time-bucket-gapfill-graph.png" alt="Screenshot of the time-series graph produced by Grafana using the `time_bucket_gapfill` hyperfunction. The graph represents the price of AMD stock in the past 30 days with an interval of 30 minutes and null values for every gap larger than 24 hours"/>
```

Comment:
This doesn't seem to be displaying in the preview.

Reply:
oh, actualy, it's the ALT text that's showing in the middle for some reason??

Reply:
I think I forgot to upload this image.